### ISSUE TITLE
StrictMode: Explicit termination method 'end' not called

### ISSUE THREAD

**Issue by ShivamPokhriyal** [firebase/firebase-android-sdk#1796]
**Firebase crashlytics version** :17.0.0

**Stack Trace**

```
A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
java.lang.Throwable: Explicit termination method 'end' not called
	at dalvik.system.CloseGuard.open(CloseGuard.java:184)
	at java.util.zip.Inflater.<init>(Inflater.java:82)
	at java.util.zip.ZipFile.getInputStream(ZipFile.java:331)
	at java.util.jar.JarFile.getInputStream(JarFile.java:390)
	at libcore.net.url.JarURLConnectionImpl.getInputStream(JarURLConnectionImpl.java:222)
	at java.net.URL.openStream(URL.java:470)
	at java.lang.ClassLoader.getResourceAsStream(ClassLoader.java:444)
	at java.lang.Class.getResourceAsStream(Class.java:1334)
	at com.google.android.gms.common.internal.LibraryVersion.getVersion(Unknown Source)
	at com.google.firebase.iid.zzt.zzb(com.google.firebase:firebase-iid@@20.1.5:61)
	at com.google.firebase.iid.zzt.zza(com.google.firebase:firebase-iid@@20.1.5:72)
	at com.google.firebase.iid.zzs.run(com.google.firebase:firebase-iid@@20.1.5)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:818)
StrictMode VmPolicy violation with POLICY_DEATH; shutting down.
```

**Comment by kevinkokomani** [comment]
Could you add more context as to how you ran into this error? Any way to reproduce this or information about your current configuration?

**Comment by ShivamPokhriyal** [comment]
I'm getting these crashes on running instrumentation tests in debug build. The tests are flaky due to this, most of the times they'll fail with the above crash. 

We've included FirebaseCrashlytics using the official documentation. I don't have much information about how to reproduce it as of now. So including all the FirebaseCrashlytics logs from the crash build here in case it helps.

```
I/FirebaseCrashlytics(18665): Initializing Crashlytics 17.0.0
I/FirebaseInitProvider(18665): FirebaseApp initialization successful
E/FirebaseInstanceId(18665): Topic sync or token retrieval failed on hard failure exceptions: AUTHENTICATION_FAILED. Won't retry the operation.
E/StrictMode(18665): A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
java.lang.Throwable: Explicit termination method 'end' not called
	at dalvik.system.CloseGuard.open(CloseGuard.java:184)
	at java.util.zip.Inflater.<init>(Inflater.java:82)
	at java.util.zip.ZipFile.getInputStream(ZipFile.java:331)
	at java.util.jar.JarFile.getInputStream(JarFile.java:390)
	at libcore.net.url.JarURLConnectionImpl.getInputStream(JarURLConnectionImpl.java:222)
	at java.net.URL.openStream(URL.java:470)
	at java.lang.ClassLoader.getResourceAsStream(ClassLoader.java:444)
	at java.lang.Class.getResourceAsStream(Class.java:1334)
	at com.google.android.gms.common.internal.LibraryVersion.getVersion(Unknown Source)
	at com.google.firebase.iid.zzt.zzb(com.google.firebase:firebase-iid@@20.1.5:61)
	at com.google.firebase.iid.zzt.zza(com.google.firebase:firebase-iid@@20.1.5:72)
	at com.google.firebase.iid.zzs.run(com.google.firebase:firebase-iid@@20.1.5)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:818)
StrictMode VmPolicy violation with POLICY_DEATH; shutting down.
```

**Comment by kevinkokomani** [comment]
Thanks for the info! A couple more things to verify:
- if you upgrade to the latest version of Crashlytics does this still occur? `implementation 'com.google.firebase:firebase-crashlytics:17.1.1'`
- if you remove Crashlytics do you still run into this?

**Comment by ShivamPokhriyal** [comment]
Hey, I had tried upgrading crashlytics to the latest version and the crash still occurred. I haven't tried removing Crashlytics yet but can give it a try.

**Comment by mrwillis21** [comment]
This looks to be related to a dependency of the Crashlytics SDK which has been upgraded in a soon-to-be-released version. If you can give us a bit to get that released and give it a shot, I suspect it'll solve the problem. I'll let you know once it's ready. :)

**Comment by mrwillis21** [comment]
Hi @ShivamPokhriyal can you try upgrading to Crashlytics 17.2.1 and let us know if you're still seeing this issue? Thanks!

**Comment by ShivamPokhriyal** [comment]
@mrwillis21 Thanks for notifying. I'll try running a couple builds using this version and will update you by the end of this week.

**Comment by ShivamPokhriyal** [comment]
@mrwillis21 Seems like the issue is resolved in `17.2.1`. Thanks for the fix.
